client/my-sites/term-tree-selector/terms.jsx

```diff
@@ -94,7 +94,8 @@ const TermTreeSelectorList = React.createClass( {
 		const forceUpdate = (
 			! isEqual( prevProps.selected, this.props.selected ) ||
 			prevProps.loading && ! this.props.loading ||
-			( ! prevProps.terms && this.props.terms )
+			( ! prevProps.terms && this.props.terms ) ||
+			( ( prevProps.terms && this.props.terms ) && ( prevProps.terms.length !== this.props.terms.length ) )
```

Comment:
Aside: I've been using [`_.size`](https://lodash.com/docs#size) when wanting to get length when array is possibly undefined (in which case returns `0`, sufficient in some use cases including this one).